library/Core/src/Juvix/Core/Common/Context/Types.hs

```diff
@@ -58,15 +56,73 @@ data Definition term ty sumRep
   | -- Signifies that this path is the current module, and that
     -- we should search the currentNameSpace from here
     CurrentNameSpace
-  deriving (Show, Generic, Eq, Data)
+  deriving (Show, Generic, Eq)
+
+data Record term ty sumRep
+  = Rec
+      { recordContents :: NameSpace.T (Definition term ty sumRep),
+        -- Maybe as I'm not sure what to put here for now
```

Comment:
explicate, why?

Reply:
As modules should always have types, however currently we don't know our module typing yet, so I just leave it as maybe typed for now... todo with more core updates